src/app/storage.js

```diff
@@ -28,12 +33,16 @@ function Storage () {
 
   this.remove = function (name) {
     window.localStorage.removeItem('sol:' + name)
+    return true
   }
 
   this.rename = function (originalName, newName) {
     var content = this.get(originalName)
-    this.set(newName, content)
+    if (!this.set(newName, content)) {
```

Comment:
Note that the IDE part is not interacting with this API at all, only through the "files API" which does this check.